File: Lesson5 Sorting/Problem1/Solutions/Solution.py
## credits to DMOJ user nathanl3
import sys
input = sys.stdin.readline

s = int(input())
n = int(input())
things = []
for i in range(n):
    b, d, c = map(int, input().split())
    things.append((b, d, c, i + 1))

def compare(x, y):
    b1, d1, c1, i1 = x
    b2, d2, c2, i2 = y
    if b1 == b2 and b1 >= s:
        return c2 - c1
    elif b1 == b2 and b1 < s:
        return d2 - d1
    else:
        return b2 - b1

things.sort(cmp=compare)

for i in range(int(input())):
    print(things[int(input()) - 1][3])

File: Lesson5 Sorting/Problem1/Solutions/Solution.cpp
// credits to DMOJ user bruce
#include <bits/stdc++.h>
using namespace std;
struct A{int b, d, c, id;}a[50002];
int N, Q, S, q;
bool cmp(A x, A y){
    if(x.b != y.b) return x.b>y.b;
    else if(x.b>=S) return x.c>y.c;
    else if(x.b<S) return x.d>y.d;
    else return x.id<y.id;
}
int main(){
    scanf("%d %d", &S, &N);
    for(int i=1; i<=N; i++){
        scanf("%d %d %d", &a[i].b, &a[i].d, &a[i].c);
        a[i].id=i;
    }
    sort(a+1, a+1+N, cmp);
    scanf("%d", &Q);
    while(Q--){
        scanf("%d", &q);
        printf("%d\n", a[q].id);
    }
}
